command/bitmarkd/main.go

```diff
@@ -290,17 +291,36 @@ func main() {
 		log.Criticalf("zmq.AuthStart: error: %s", err)
 		exitwithstatus.Message("zmq.AuthStart: error: %s", err)
 	}
+
 	if masterConfiguration.DNSPeerOnly {
-		err = p2p.Initialise(&masterConfiguration.Peering, version, masterConfiguration.Fastsync, p2p.DnsOnly)
+		err = p2p.Initialise(&masterConfiguration.Peering, version, p2p.DnsOnly)
 	} else {
-		err = p2p.Initialise(&masterConfiguration.Peering, version, masterConfiguration.Fastsync, p2p.UsePeers)
+		err = p2p.Initialise(&masterConfiguration.Peering, version, p2p.UsePeers)
 	}
 
 	if nil != err {
 		log.Criticalf("p2p initialise error: %s", err)
 		exitwithstatus.Message("p2p initialise error: %s", err)
 	}
 	defer p2p.Finalise()
+
+	for wait := 0; wait < 3; wait++ { // consensus package depended on p2p.Node
+		if nil == p2p.GlobalP2PNode() {
```

Comment:
I am curious about this, the function of `p2p.GlobalP2PNode` is to return internal variable:
```
func GlobalP2PNode() *Node {
	return &globalData
}
```

Does it necessary to access internal variable?

If it's yes, then why not change internal variable into global accessible variable? (`globalData` => `GlobalData`)

Reply:
The consensus package needs to access the node. The difference is that if make globaldata public, then a user can do p2p.GlobalNode = NewNode() but if it open only "get function". Then a user can only use the return node.

Reply:
If the purpose is to protect from modification, then return pointer cannot have it. Anyone can use following code to change content:

```
p := p2p.P2PNode()
p.Registers = xxxx
```

Return pointer cannot protect others from changing content. I would suggest using interface instead of exposing variables.

But that is just a suggestion.